internal/dnsanalyzer/dnsanalyzer.go

```diff
@@ -0,0 +1,108 @@
+package dnsanalyzer
+
+import (
+	"net"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+type rawLookup struct {
+	query *layers.DNS
+	reply *layers.DNS
+}
+
+type DNSAnalyzer struct {
+	lookups     map[uint16]*rawLookup
+	ipHostnames map[string]map[string]struct{}
```

Comment:
Let's make a `type empty struct{}` to make this more readable.

Reply:
Good idea. Done!